src/main/java/de/zalando/aruha/nakadi/partitioning/OrderingKeyFieldsPartitioningStrategy.java

```diff
@@ -0,0 +1,72 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import static java.lang.Math.abs;
+
+public class OrderingKeyFieldsPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final String[] partitions) throws InvalidOrderingKeyFieldsException {
+        final List<String> orderingKeyFields = eventType.getOrderingKeyFields();
+        if (orderingKeyFields.isEmpty()) {
```

Comment:
I don't see it as a responsibility of this strategy to fall back to another strategy if it cannot be applied. If someone tries to apply this strategy although it's not applicable, I see it as an error. Don't you think so?